zebra-consensus/src/block.rs

```diff
@@ -100,14 +100,14 @@ where
             if hash > difficulty_threshold {
                 Err("Block failed the difficulty filter: hash must be less than or equal to the difficulty threshold.")?;
             }
-            block.header.is_equihash_solution_valid()?;
+            check::is_equihash_solution_valid(&block.header)?;
```

Comment:
This change will conflict with some of the error cleanup in #1012, but they should be easy enough to merge.

Reply:
Will keep an eye on that PR.